source/features/mute-users.tsx

```diff
@@ -0,0 +1,134 @@
+import React from 'dom-chef';
+import select from 'select-dom';
+import delegate, {DelegateEvent} from 'delegate-it';
+import features from '../libs/features';
+import {appendBefore} from '../libs/dom-utils';
+import optionsStorage, {Options} from '../options-storage';
+import {getUsername} from '../libs/utils';
+
+let currentUser: string;
+
+async function getMutedUsers(): Promise<string[]> {
+	return (await optionsStorage.getAll() as Options).mutedUsers.split(/\s+/);
+}
+
+async function setMutedUsers(mutedUsers: string[]): Promise<void> {
+	return optionsStorage.set({mutedUsers: mutedUsers.join(' ')});
+}
+
+function minimizeComment(comment: HTMLElement): void {
+	if (select.exists('.js-targetable-comment[id^="issue-"]', comment)) {
+		return;
+	}
+
+	select('.minimized-comment', comment)!.classList.remove('d-none');
+	select('.minimized-comment .Details-element', comment)!.removeAttribute('open');
+	select('.unminimized-comment', comment)!.classList.add('d-none');
+}
+
+function unminimizeComment(comment: HTMLElement): void {
+	if (select.exists('.js-targetable-comment[id^="issue-"]', comment)) {
+		return;
+	}
+
+	select('.minimized-comment', comment)!.classList.add('d-none');
+	select('.minimized-comment .Details-element', comment)!.setAttribute('open', 'true');
+	select('.unminimized-comment', comment)!.classList.remove('d-none');
+}
+
+async function onMuteUnmuteClick(event: React.MouseEvent<HTMLButtonElement>): Promise<void> {
+	let mutedUsers = await getMutedUsers();
+
+	const comment = (event.target as HTMLElement).closest('.js-comment-container')!;
+	const viewportOffset = comment.getBoundingClientRect().top;
+	const user = select('.author', comment)!.textContent!;
+
+	if (mutedUsers.includes(user)) {
+		mutedUsers = mutedUsers.filter(mutedUser => mutedUser !== user);
+	} else {
+		mutedUsers.push(user);
+	}
+
+	await setMutedUsers(mutedUsers);
+
+	const avatars = select.all(`.js-discussion .js-comment-container .avatar[alt="@${user}"`);
+	const comments = avatars.map(avatar => avatar.closest('.js-comment-container') as HTMLElement);
+
+	for (const comment of comments) {
+		if (mutedUsers.includes(user)) {
+			minimizeComment(comment);
+		} else {
+			unminimizeComment(comment);
+		}
+	}
+
+	requestAnimationFrame(() => {
+		const newOffset = comment.getBoundingClientRect().top;
+		window.scrollBy(0, newOffset - viewportOffset);
+	});
```

Comment:
Previous review 👆 

Everything after setMutedComments should not be here. Just call minimizeMutedUserComments

Reply:
From https://github.com/sindresorhus/refined-github/pull/2146#discussion_r295721161.

Reply:
I understand the need for the function and I’m saying that it should be part of `minimizeMutedUserComments`

Reply:
In `minimizeMutedUserComments`, there is no concept of a `comment` on which the original event was triggered. That exists only inside `onMuteUnmuteClick`.

Reply:
Yeah. "Comment" would be "whatever is centered on screen right now", e.g. `document.elementFromPoint(innerWidth/2, innerHeight/2)`

There's a library (apparently written for GitHub itself) that gets [the element the mouse is hovering](https://github.com/josh/scroll-anchoring/blob/master/scroll-anchoring.js#L61), but I think that doesn't work well with fixed elements (sidebar, header)

Browsers have a native concept of "scroll anchoring" recently added by both Chrome and Firefox, but I don't think it works in all cases, so probably we can't rely on it.

Reply:
If you wrap it into its own `scroll-anchoring.ts` function it can be used by other features as well: https://github.com/sindresorhus/refined-github/pull/2087/files#diff-7259b235bc8829b04689968edcc74f94R58

Maybe `anchorScroll(fn: function, anchor?: Node)`